x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/query_signals_index_schema.ts

```diff
@@ -9,4 +9,7 @@ import Joi from 'joi';
 export const querySignalsSchema = Joi.object({
   query: Joi.object(),
   aggs: Joi.object(),
+  size: Joi.number(),
+  track_total_hits: Joi.boolean(),
+  _source: Joi.array().items(Joi.string()),
```

Comment:
👍 

You can do `size: Joi.number().min(1)` if you want to prevent people from sending in -1 or 0.

FWIW, there is a test file for this here:
```ts
x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/query_signals_index_schema.test.ts
```

That's easy to follow along to add new validation unit test cases.

Reply:
We want the zero because sometimes you just want `aggs`